src/hwProvider.ts

```diff
@@ -13,12 +13,23 @@ import { IHWWalletApp } from "./interface";
 import { compareVersions } from "./versioning";
 
 export class HWProvider {
-    hwApp?: IHWWalletApp;
-    addressIndex: number = 0;
+    constructor(public _addressIndex: number = 0, public _hwApp?: IHWWalletApp) {}
 
-    constructor() {
+    public get addressIndex(): number {
+        return this._addressIndex;
     }
 
+    public set addressIndex(addressIndex: number) {
+        this._addressIndex = addressIndex;
+    }
+
+    public get hwApp(): IHWWalletApp | undefined {
+        return this._hwApp;
+    }
+
+    public set hwApp(hwApp: IHWWalletApp | undefined) {
```

Comment:
Actually, this should never be set by client code, I think.

Perhaps - simply receive the `hwApp` in the constructor - as you did, without the setter?